lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/auth/OAuthClientDaoImpl.java

```diff
@@ -222,8 +221,19 @@ public boolean checkOAuthClientSecretHash(String clientId, String secretHash) {
 
 	@Override
 	public byte[] getSecretSalt(String clientId) {
-		String secretHash = jdbcTemplate.queryForObject(CLIENT_SECRET_HASH_SQL_SELECT, String.class, clientId);
-		if (secretHash==null) throw new NotFoundException("OAuth client (" + clientId + ") does not exist");
+		
+		String secretHash;
+		
+		try {
+			secretHash = jdbcTemplate.queryForObject(CLIENT_SECRET_HASH_SQL_SELECT, String.class, clientId);
+		} catch (EmptyResultDataAccessException e) {
+			throw clientNotFoundException(clientId);
+		}
+		
+		if (secretHash == null) {
+			throw new NotFoundException("OAuth client (" + clientId + ") not initialized");
```

Comment:
This error message is an improvement but I'd go further and replace 'not initialized' with something like 'does not have a secret defined'